src/server_manager/web_app/app.ts

```diff
@@ -1045,6 +1063,13 @@ export class App {
       this.appRoot.showNotification(this.appRoot.localize('saved'));
       serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(limit);
       this.refreshTransferStats(this.selectedServer, serverView);
+      await sleep(HATS_DIALOG_DELAY_MS);
```

Comment:
Use `setTimeout` instead to not block the thread

Reply:
`sleep` uses `setTimeout` internally. I prefer to reduce nesting.